pkg/resources/statefulsets/minio-statefulset.go

```diff
@@ -134,24 +118,33 @@ func minioEnvironmentVars(t *miniov2.Tenant, wsSecret *v1.Secret, hostsTemplate
 	}
 	// Check if any domains are configured
 	if t.HasMinIODomains() {
-		domains = append(domains, t.Spec.Features.Domains.Minio...)
+		domains = append(domains, t.GetDomainHosts()...)
 	}
-	// tell MinIO about all the domains meant to hit it
-	if len(domains) > 0 {
+	// tell MinIO about all the domains meant to hit it if they are not passed manually via .spec.env
+	if !t.HasEnv("MINIO_DOMAIN") && len(domains) > 0 {
 		envVars = append(envVars, corev1.EnvVar{
 			Name:  "MINIO_DOMAIN",
 			Value: strings.Join(domains, ","),
 		})
 	}
+	// If no specific server URL is specified we will specify the internal k8s url, but if a list of domains was
+	// provided we will use the first domain.
+	if !t.HasEnv("MINIO_SERVER_URL") {
+		serverURL := t.MinIOServerEndpoint()
+		if t.HasMinIODomains() {
+			serverURL = t.Spec.Features.Domains.Minio[0]
+		}
+		envVars = append(envVars, corev1.EnvVar{
+			Name:  "MINIO_SERVER_URL",
+			Value: serverURL,
+		})
+	}
+
 	// Set the redirect url for console
 	if t.HasConsoleDomains() {
-		schema := "http"
-		if t.TLS() {
-			schema = "https"
-		}
 		envVars = append(envVars, corev1.EnvVar{
 			Name:  "MINIO_BROWSER_REDIRECT_URL",
-			Value: fmt.Sprintf("%s://%s", schema, t.Spec.Features.Domains.Console),
```

Comment:
isn't this a breaking change?

Reply:
A little, I'm hoping this change is not too disruptive, alternatively I could attach the scheme if it's not present.

Reply:
That is needed IMO, we can't break existing deployments.

Reply:
addressed, made it so if the schema is missing I add it, just like before